File: WorkOffice.Contracts/Models/NHS/DiagnosticModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;
using System.Text.Json.Serialization;

namespace WorkOffice.Contracts.Models
{
   public class DiagnosticModel
    {
        public int DiagnosticId { get; set; }
        public int PatientId { get; set; }
        public string PatientName { get; set; }
        public int SpecialtyId { get; set; }
        public string Specialty { get; set; }
        public string Problem { get; set; }
        public DateTime DTD { get; set; }
        public string Status { get; set; }
        public string ConsultantName { get; set; }
        public bool? Active { get; set; }
        [JsonIgnore]
        public string CurrentUserName { get; set; }
    
    }
}

File: WorkOffice.Web/ClientApp/src/app/nhs/diagnostic/diagnostic.model.ts

export interface DiagnosticModel {
  diagnosticId: number;
  patientId: number;
  patientName: string
  specialtyId: number;
  specialty: string;
  dtd: Date;
  problem: string;
  status: string;
  consultantName: string;
}

export interface DiagnosticResultModel {
  diagnosticResultId: number;
  diagnosticId: number;
  problem: string;
  consultantName: string;
  documentName: string;
  documentExtension: string;
  documentFile: string;
  testResultDate:Date;
  specialityId: number;
  dateUploaded: Date;
}

